t/lib/t/MusicBrainz/Server/Data/Role/GIDEntityCache.pm

```diff
@@ -189,24 +269,25 @@ test 'Cache is transactional (MBS-7241)' => sub {
             'cache is not repopulated after concurrent get_by_gid');
     };
 
-    Sql::run_in_transaction(sub {
-        $c1->model('Artist')->delete(3);
-    }, $c1->sql);
-
-    my $artist = $c1->model('Artist')->get_by_id(3);
-    ok(!defined $artist,
-        'get_by_id returns undef after ' .
-        'database deletion commits, and after cache deletion');
-
-    $c1->sql->auto_commit(1);
-    $c1->sql->do(<<~'SQL');
-        DELETE FROM artist WHERE id = 3;
-        DELETE FROM deleted_entity WHERE gid = '31456bd3-0e1a-4c47-a5cc-e147a42965f2';
-        SQL
+    try {
+        Sql::run_in_transaction(sub {
+            $c1->model('Artist')->delete(3);
+        }, $c1->sql);
+
+        my $artist = $c1->model('Artist')->get_by_id(3);
+        ok(!defined $artist,
+            'get_by_id returns undef after ' .
+            'database deletion commits, and after cache deletion');
```

Comment:
a

Reply:
To follow the above suggestion, and since database deletion commits after cache deletion anyway:
```suggestion
        ok(!defined $artist,
            '(c.) get_by_id returns undef after ' .
            'database deletion commits');
```

Similarly, (a.) and (b.) could be inserted into their respective test message.